### PR TITLE
ztp: refactor getExtraManifest so it also processes workload partition cr

### PR DESCRIPTION
**Pull request by pixelsoccupied**
This should help streamline new features such a filter and merge which can be called inside the `getExtraManifest` function

### CODE REVIEW

File: ztp/siteconfig-generator/siteConfig/siteConfigBuilder.go [hunk 1]
```diff
@@ -404,6 +389,20 @@ func (scbuilder *SiteConfigBuilder) getExtraManifest(dataMap map[string]interfac
 		}
 	}
 
+	// Adding workload partitions MC only for SNO clusters.
```
[the remark this answers is not in the document]
Comment (pixelsoccupied) [reply ztp/siteconfig-generator/siteConfig/siteConfigBuilder.go]: moved

**Review by serngawy — COMMENTED** [review]
sounds good, will wait till ci job done then approve it

**Review by imiller0 — APPROVED** [review]
/lgtm

**Pull request merged**